### ISSUE TITLE
Refactor D2L notebooks

### ISSUE THREAD

**Issue by patel-zeel** [probml/pyprobml#801]
## Dashboard

| Notebook | Beautify | Jaxify |
|----------|----------|--------|
| [naive_bayes_mnist_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/09/naive_bayes_mnist_jax.ipynb) | PR | PR |
| [image_augmentation_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/19/image_augmentation_jax.ipynb) | PR | PR |
| [finetune_cnn_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/19/finetune_cnn_jax.ipynb) | PR | PR |
| [skipgram_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/20/skipgram_jax.ipynb) | PR | PR |
| [nmt_attention_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/15/nmt_attention_jax.ipynb) | PR | PR |
| [resnet_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/14/resnet_jax.ipynb) | PR | PR |
| [lenet_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/14/lenet_jax.ipynb) | PR | PR |
| [batchnorm_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/14/batchnorm_jax.ipynb) | PR | PR |
| [nmt_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/15/nmt_jax.ipynb) | PR | PR |
| [multi_head_attention_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/15/multi_head_attention_jax.ipynb) | PR | PR |
| [gru_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/15/gru_jax.ipynb) | PR | PR |
| [cnn1d_sentiment_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/15/cnn1d_sentiment_jax.ipynb) | PR | PR |
| [entailment_attention_mlp_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/15/entailment_attention_mlp_jax.ipynb) | PR | PR |
| [conv2d_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/14/conv2d_jax.ipynb) | PR | PR |
| [attention_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/15/attention_jax.ipynb) | PR | PR |
| [rnn_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/15/rnn_jax.ipynb) | PR | PR |
| [densenet_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/14/densenet_jax.ipynb) | PR | PR |
| [positional_encoding_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/15/positional_encoding_jax.ipynb) | PR | PR |
| [transformers_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/15/transformers_jax.ipynb) | PR | PR |
| [bert_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/15/bert_jax.ipynb) | PR | PR |
| [rnn_sentiment_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/15/rnn_sentiment_jax.ipynb) | PR | PR |
| [word_analogies_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/20/word_analogies_jax.ipynb) | PR | PR |
| [multi_gpu_training_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/13/multi_gpu_training_jax.ipynb) | PR | PR |
| [lstm_jax](https://github.com/probml/pyprobml/tree/master/notebooks/book1/15/lstm_jax.ipynb) | PR | PR |

cc @nalzok.

**Comment by murphyk** [comment]
Closing since no longer a priority.
